Source/PeleLMeX_Diffusion.cpp

```diff
@@ -1299,49 +1323,92 @@ PeleLM::differentialDiffusionUpdate(
 
     auto* ldata_p = getLevelDataPtr(lev, AmrNewTime);
 
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
-#endif
-    for (amrex::MFIter mfi(ldata_p->state, amrex::TilingIfNotGPU());
-         mfi.isValid(); ++mfi) {
-      const amrex::Box& bx = mfi.tilebox();
-      amrex::FArrayBox DummyFab(bx, 1);
-      auto const& rhoY = ldata_p->state.array(mfi, FIRSTSPEC);
-      auto const& dhat = diffData->Dhat[lev].const_array(mfi);
-      auto const& force = advData->Forcing[lev].const_array(mfi, 0);
-      auto const& dwbar = (m_use_wbar != 0)
-                            ? diffData->Dwbar[lev].const_array(mfi)
-                            : DummyFab.const_array();
-      auto const& dT = (m_use_soret != 0) ? diffData->DT[lev].const_array(mfi)
-                                          : DummyFab.const_array();
-      auto const& aux =
-        (m_nAux > 0) ? ldata_p->auxiliaries.array(mfi, 0) : DummyFab.array();
-      auto const& dhat_aux = (m_nAux > 0)
-                               ? diffData->Dhat_aux[lev].const_array(mfi)
-                               : DummyFab.const_array();
-      auto const& force_aux = (m_nAux > 0)
-                                ? advData->Forcing_aux[lev].const_array(mfi, 0)
-                                : DummyFab.const_array();
-      const auto nAux = m_nAux;
-      const auto dt = m_dt;
-      const auto use_wbar = m_use_wbar;
-      const auto use_soret = m_use_soret;
+    auto const& state_ma = ldata_p->state.arrays();
+    auto const& dhat_ma = diffData->Dhat[lev].const_arrays();
+    auto const& force_ma = advData->Forcing[lev].const_arrays();
+
+    auto const& dwbar_ma =
+      (m_use_wbar != 0) ? diffData->Dwbar[lev].const_arrays() : dhat_ma;
+    auto const& dT_ma =
+      (m_use_soret != 0) ? diffData->DT[lev].const_arrays() : dhat_ma;
+    auto const& aux_ma =
+      (m_nAux > 0) ? ldata_p->auxiliaries.arrays() : state_ma;
+    auto const& dhat_aux_ma =
+      (m_nAux > 0) ? diffData->Dhat_aux[lev].const_arrays() : dhat_ma;
+    auto const& force_aux_ma =
+      (m_nAux > 0) ? advData->Forcing_aux[lev].const_arrays() : dhat_ma;
+
+    if (m_use_wbar != 0 && m_use_soret != 0) {
       amrex::ParallelFor(
-        bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-          for (int n = 0; n < NUM_SPECIES; n++) {
-            rhoY(i, j, k, n) = force(i, j, k, n) + dt * dhat(i, j, k, n);
-            if (use_wbar != 0) {
-              rhoY(i, j, k, n) -= dt * dwbar(i, j, k, n);
-            }
-            if (use_soret != 0) {
-              rhoY(i, j, k, n) -= dt * dT(i, j, k, n);
-            }
+        ldata_p->state,
```

Comment:
Same comment for breaking up and increasing parallelism. You don't need to do these now btw. Another PR is probably better